### PR TITLE
Update to new API

### PR DESCRIPTION
**Pull request by willtebbutt**
See [ChainRulesCore](https://github.com/JuliaDiff/ChainRulesCore.jl/pull/129) for related PR

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -4,6 +4,7 @@ version = "0.3.4"
 
 [deps]
 ChainRulesCore = "d360d2e6-b24c-11e9-a2a3-2a2ae2dbcce4"
+ChainRulesTestUtils = "cdddcdb0-9152-4a09-a978-84456f9df70a"
```
Comment (nickrobinson251) [inline Project.toml]: Should be test-only dep?

File: Project.toml [hunk 2]
```diff
@@ -12,7 +13,7 @@ Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 [compat]
 ChainRulesCore = "0.6.1"
```
Comment (nickrobinson251) [inline Project.toml]: Need to change this to `0.7.0`

File: Project.toml [hunk 3]
```diff
@@ -12,7 +13,7 @@ Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
...
 ChainRulesTestUtils = "0.1.3"
-FiniteDifferences = "^0.7"
+FiniteDifferences = "0.7"
 Reexport = "0.2"
 Requires = "0.5.2, 1"
 julia = "^1.0"
```
Comment (nickrobinson251) [inline Project.toml]: since we're doing this for FiniteDifferences

```suggestion
julia = "1"
```

File: Project.toml (hunk 1 above)
Comment (nickrobinson251) [reply Project.toml]:
```suggestion
```

File: Project.toml [hunk 4]
```diff
@@ -12,7 +13,7 @@ Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 [compat]
 ChainRulesCore = "0.6.1"
 ChainRulesTestUtils = "0.1.3"
```
Comment (nickrobinson251) [inline Project.toml]: Needs bumping to match https://github.com/JuliaDiff/ChainRulesTestUtils.jl/pull/17

```suggestion
ChainRulesTestUtils = "0.2"
```

File: Project.toml (hunk 2 above)
Comment (willtebbutt) [reply Project.toml]:
```suggestion
ChainRulesCore = "0.7.0-DEV"
```

**Comment by willtebbutt** [comment]
@nickrobinson251 let me know if you're happy for this to be merged now.

File: .travis.yml [hunk 5]
```diff
@@ -10,7 +10,6 @@ julia:
   - nightly
```
Comment (oxinabox) [inline .travis.yml]:
```suggestion
  - 1.4
```

**Comment by willtebbutt** [comment]
> It is worrying how few tests needed to be updated

Yeah. This is what caused this slip up with the API in the first place -- we simply didn't have the range of things implemented to catch the problem. I think that the right place to resolve this issue is the core though -- I've implemented extra tests to prevent API regression, but presumably there are other things we're not handling properly (kwargs, for example).

**Comment by willtebbutt** [comment]
@oxinabox @nickrobinson251 anyone able to reproduce the errors locally? Tests are passing for me on 1.3.1. Have checked versions, and I appear to be using the same as CI is using...

**Comment by oxinabox** [comment]
Oh is this: https://github.com/JuliaDiff/ChainRules.jl/issues/149 ?

I think we had travis disabled on 1.3 for reasons because it was just broken even though it worked fine in 1.3 on everyones local machines

**Comment by willtebbutt** [comment]
Ahh I see. Will re-disable and merge once tests pass.

**Pull request merged**